cpp/benchmarks/io/text/multibyte_split.cpp

```diff
@@ -103,22 +103,23 @@ static void bench_multibyte_split(nvbench::state& state)
   auto const delim_factor = static_cast<double>(delim_percent) / 100;
   auto device_input       = create_random_input(file_size_approx, delim_factor, 0.05, delim);
   auto host_input         = thrust::host_vector<char>(device_input.size());
-  auto const host_string  = std::string(host_input.data(), host_input.size());
 
   cudaMemcpyAsync(host_input.data(),
                   device_input.data(),
                   device_input.size() * sizeof(char),
                   cudaMemcpyDeviceToHost,
                   cudf::default_stream_value);
 
+  cudaDeviceSynchronize();
+
   auto const temp_file_name = random_file_in_dir(temp_dir.path());
 
   {
     auto temp_fostream = std::ofstream(temp_file_name, std::ofstream::out);
     temp_fostream.write(host_input.data(), host_input.size());
   }
 
-  cudaDeviceSynchronize();
+  auto const host_string = std::string(host_input.data(), host_input.size());
 
   auto source = [&] {
     switch (source_type) {
```

Comment:
One or two sources are created and left unused in some benchmarks, depending on the source type. 
We can avoid more writes/allocations/copies:
1. Declare `host_string` as empty and only resize/copy if source type is not `device`;
2. Only write input to the file i the source type is `file`.

Reply:
great suggestions, I incorporated both